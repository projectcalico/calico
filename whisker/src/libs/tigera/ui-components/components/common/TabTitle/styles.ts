export const tabTitleTextStyles = (hasNodata: boolean) => ({
    fontSize: 'xs',
    fontWeight: 600,
    color: hasNodata ? 'tigeraGrey.500' : 'inherit',
});

export const tagStyles = {
    paddingInlineEnd: 2,
    paddingInlineStart: 2,
    minWidth: 'unset',
    alignItems: 'center',
    ml: 2,
    fontSize: 'xs',
};
