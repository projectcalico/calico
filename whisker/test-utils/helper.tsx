import React, { ReactElement } from 'react';
import { render, renderHook, RenderOptions } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { theme } from '../src/theme';
import { MemoryRouter } from 'react-router';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';

const queryClient = new QueryClient({
    defaultOptions: {
        queries: {
            retry: false,
        },
    },
});

const queryClientProvider = ({ children }: any) => (
    <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

export const renderHookWithQueryClient = (hook: () => unknown) =>
    renderHook(hook, { wrapper: queryClientProvider });

const Providers = ({ children }: { children: React.ReactNode }) => {
    return <ChakraProvider theme={theme}>{children}</ChakraProvider>;
};

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options });

export const renderWithRouter = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'wrapper'> & {
        routes: string[];
    },
) =>
    customRender(
        <MemoryRouter initialEntries={options?.routes}>{ui}</MemoryRouter>,
        options,
    );

export * from '@testing-library/react';

export { customRender as render };
