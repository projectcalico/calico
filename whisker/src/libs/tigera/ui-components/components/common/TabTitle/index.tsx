import { Box } from '@chakra-ui/react';
import type { SystemStyleObject, HTMLChakraProps } from '@chakra-ui/react';
import { tabTitleTextStyles, tagStyles } from './styles';
import Tag from '../Tag';

export interface TabTitleProps extends HTMLChakraProps<'div'> {
    isSelected?: boolean;
    hasNoData: boolean;
    badgeCount?: number;
    title: string;
    sx?: SystemStyleObject;
}

const TabTitle = ({
    isSelected = true,
    hasNoData,
    badgeCount,
    sx,
    title,
    ...rest
}: TabTitleProps) => {
    return (
        <Box
            data-role='tabTitle'
            sx={{ ...tabTitleTextStyles(hasNoData), ...sx }}
            {...rest}
        >
            {title}

            <Tag sx={tagStyles} isActive={isSelected}>
                {badgeCount}
            </Tag>
        </Box>
    );
};

export default TabTitle;
