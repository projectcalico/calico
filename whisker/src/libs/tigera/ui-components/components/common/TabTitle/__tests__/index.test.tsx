import { render } from '@/test-utils/helper';
import TabTitle from '..';

const defaultProps = {
    isSelected: true,
    hasNoData: false,
    badgeCount: 10,
    title: 'some title',
} as any;

describe('<TabTitle', () => {
    it('renders TabTitle -- happy path', () => {
        const { asFragment } = render(<TabTitle {...defaultProps} />);

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders TabTitle -- alt path when has no data', () => {
        const { asFragment } = render(
            <TabTitle {...defaultProps} hasNoData={true} isSelected={false} />,
        );

        expect(asFragment()).toMatchSnapshot();
    });

    it('renders TabTitle -- alt path when not selected', () => {
        const { asFragment } = render(
            <TabTitle {...defaultProps} hasNoData={false} isSelected={false} />,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
