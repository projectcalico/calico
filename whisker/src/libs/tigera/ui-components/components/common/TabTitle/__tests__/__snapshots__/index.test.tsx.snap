// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TabTitle renders TabTitle -- alt path when has no data 1`] = `
<DocumentFragment>
  <div
    class="css-14nqeq7"
    data-role="tabTitle"
  >
    some title
    <span
      class="css-kg5w72"
    >
      10
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<TabTitle renders TabTitle -- alt path when not selected 1`] = `
<DocumentFragment>
  <div
    class="css-ltvj7l"
    data-role="tabTitle"
  >
    some title
    <span
      class="css-kg5w72"
    >
      10
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<TabTitle renders TabTitle -- happy path 1`] = `
<DocumentFragment>
  <div
    class="css-ltvj7l"
    data-role="tabTitle"
  >
    some title
    <span
      class="css-jiyp2n"
    >
      10
    </span>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
